scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,330 @@
+        if (
+            self.ad_tech_platform_bid_requests_processed_billion_per_month
+            and self.bid_requests_processed_billion_per_month
+        ):
+            ratio = (
+                self.ad_tech_platform_bid_requests_processed_billion_per_month
+                / self.bid_requests_processed_billion_per_month
+            )
+            log_result("external request ratio", f"{ratio * 100.0:.1f}%", depth)
+            return ratio
+        return self.get_decimal_fraction_of_bid_requests_processed_from_ad_tech_platforms()
+
+    def comp_bid_request_size_gb(self) -> float:
+        return self.bid_request_size_in_bytes / 1024 / 1024 / 1024
+
+    def comp_data_transfer_emissions_g_co2e_per_bid_request(self, depth: int) -> float:
+        # TODO why isn't this divided by the number of bid requests
+        data_transfer_emissions_g_co2e_per_month = (
+            self.get_data_transfer_emissions_g_co2e_per_month()
+        )
+        if data_transfer_emissions_g_co2e_per_month:
+            return data_transfer_emissions_g_co2e_per_month
+
+        data_transfer_emissions_g_co2e_per_bid_request = (
+            self.comp_external_request_ratio(depth - 1)
+            * self.comp_bid_request_size_gb()
+            * self.server_to_server_emissions_g_co2e_per_gb
+        )
+        log_result(
+            "data transfer emissions g co2e per bid request",
+            f"{data_transfer_emissions_g_co2e_per_bid_request:.8f}",
+            depth,
+        )
+        return data_transfer_emissions_g_co2e_per_bid_request
+
+    def comp_server_emissions_g_co2e_per_bid_request(
+        self,
+        depth: int,
+    ) -> float:
+        server_emissions_g_co2e_per_bid_request = (
+            self.get_server_allocation_decimal_fraction()
+            * self.get_server_emissions_g_co2e_per_month()
+            * self.get_servers_processing_bid_requests_decimal_fraction()
+        ) / self.get_bid_requests_processed_per_month()
+
+        log_result(
+            "server emissions g co2e per bid request",
+            f"{server_emissions_g_co2e_per_bid_request:.6f}",
+            depth,
+        )
+        return server_emissions_g_co2e_per_bid_request
+
+    def comp_primary_emissions_g_co2e_per_bid_request(
+        self,
+        depth: int,
+    ) -> float:
+        data_transfer_emissions_g_co2e_per_bid_request = (
+            self.comp_data_transfer_emissions_g_co2e_per_bid_request(depth - 1)
+        )
+
+        server_emissions_g_co2e_per_bid_request = self.comp_server_emissions_g_co2e_per_bid_request(
+            depth - 1
+        )
+        primary_emissions_g_co2e_per_bid_request = (
+            self.corporate_emissions_g_co2e_per_bid_request
+            + data_transfer_emissions_g_co2e_per_bid_request
+            + server_emissions_g_co2e_per_bid_request
+        )
+        log_result(
+            "primary emissions g co2e per bid request",
+            f"{primary_emissions_g_co2e_per_bid_request:.6f}",
```

Comment:
Updated this to be `g` not `mt` which is is currently https://github.com/scope3data/methodology/blob/main/scope3_methodology/adtech_model.py#L181

This was wrong in the existing code
```
primary_emissions_per_bid_request = (
        corporate_emissions_per_bid_request                   -> g 
        + data_transfer_emissions_per_bid_request         -> mt or g 
        + server_emissions_per_bid_request                     -> g 
    )
```

making sure we are in the same units